src/openvic-simulation/interface/GUI.hpp

```diff
@@ -0,0 +1,230 @@
+#pragma once
+
+#include "openvic-simulation/interface/GFX.hpp"
+
+namespace OpenVic {
+	class UIManager;
+}
+namespace OpenVic::GUI {
+	class Scene;
+
+	class Element : virtual public Named<UIManager const&> {
```

Comment:
macOS Clang still does not support something with a virtual base class containing a constexpr member function.

Reply:
Nah not really, since writing that code I've realised that windows and the top level of GUI files are less similar than I thought, so it shouldn't be an issue to restructure them.